AppController/djinn.rb

```diff
@@ -5224,62 +5226,65 @@ def scale_down_instances
           # Right now, only the autoscaled machines are started with just the
           # compute role, so we check specifically for that during downscaling
           # to make sure we only downscale the new machines added.
-          node_to_remove = nil
           if node.jobs == ['compute']
             Djinn.log_info("Removing node #{node}")
-            node_to_remove = node
+            nodes_to_remove << node
           end
-
-          num_terminated = terminate_node_from_deployment(node_to_remove)
-          num_scaled_down += num_terminated
         }
+        terminate_nodes_from_deployment(nodes_to_remove)
       }
 
       # Make sure we have the proper list of blobservers configured.
       update_blob_servers
     }
   end
 
-  # Removes the specified node from the deployment and terminates
+  # Removes the specified nodes from the deployment and terminates
   # the instance from the cloud.
   #
   # Args:
-  #   node_to_remove: A node instance, to be terminated and removed
-  #     from this deployment.
-  def terminate_node_from_deployment(node_to_remove)
-    if node_to_remove.nil?
-      Djinn.log_warn("Tried to scale down but couldn't find a node to remove.")
+  #   nodes_to_remove: A node instance, or an Array of nodes to be
+  #                    terminated and removed from this deployment.
+  def terminate_nodes_from_deployment(nodes_to_remove)
+    if nodes_to_remove.nil? || nodes_to_remove.emtpy?
+      Djinn.log_info("Tried to scale down but couldn't find a node to remove.")
       return 0
     end
 
+    # Let's get the list of IDs of the instances to terminate.
+    instances_id = []
+    nodes_to_remove.each{ |node| instances_id << node.instance_id }
```

Comment:
`instance_ids` would be a more idiomatic name. This is also a good fit for map:
`instance_ids = nodes_to_remove.map{ |node| node.instance_id }`